packages/dev-server-storybook/test/fixtures/storybook-config-dir/main.js

```diff
@@ -0,0 +1,3 @@
+export default {
+  stories: ['../stories/**/*.stories.@(js|jsx|ts|tsx)'],
+};
```

Comment:
it probably works because you added `ts-node/esm` which changed the behavior of NodeJS require internals, but I don't think it's a good idea to do this, because then you are not testing how the majority of people run dev-server-storybook/Storybook6